### ISSUE TITLE
Option to add admins or a default set of roles/groups to new spaces

### ISSUE THREAD

**Issue by dwhite5914** [owncloud/ocis#10639]
## Is your feature request related to a problem? Please describe.
In our lab, we have multiple admins who need to be able to access all spaces in the lab. The current OCIS implementation allows admins to see all spaces under "Admin Settings" but they are not automatically given access to all spaces and don't seem to have access to add themselves to a space that was created by another admin. This slows us down since we always need to add the "admins" LDAP group to every new space we create.

## Describe the solution you'd like
As an admin, I would like the ability to configure admins (users with the "admin" role) to be automatically added as managers of all spaces existing and new. This could be achieved via configuration file, environment variable, or perhaps even just be the default behavior if it makes sense. This could also take the form of a default set of roles, groups, or both that will be automatically added to new spaces.

## Describe alternatives you've considered
While I prefer the ability to have admins or a default set of roles/groups automatically added as managers of all existing/new spaces, at the very least, it would be nice if admins could add themselves via the "Admin Settings" to spaces that were created by other admins.

## Additional context
This should apply generally, but I am using a three container podman setup with OCIS server, OCIS Collaboration Server, and OnlyOffice Document Server.

**Comment by micbar** [comment]
@dwhite5914 Thanks for opening this proposal.

## Where we came from

The spaces concept has a long history. A lot of ownCloud 10 users and admins gathered requirements during many years using owncloud and came up with some key aspects which we transferred into the spaces concept.

1. Division of Powers - Admins should be able to do admin tasks **without** having potentially access to all data on the platform.
2. Decentralized ownership - no user transfer of data is needed when a person leaves the organisation.
3. Self Service - No admin should be needed to let workgroups organize themselves in their "Collaboration areas".
4. Full Transparency - **Every** access given to people should be visible to all space members.

So the first aspect has driven the decision, that no Admin should be ever able to add himself to a space. This is only possible via an invite from the existing space managers.

Your proposal suggests a "default assigned group" to all new spaces like an "the admins are watching us" 😏 which is a bit against the idea of "self organized working areas".

On the other hand, together with the transparency aspect, it could be a valid use case, as long as the "administrators" are well-known to the other users.

IMHO this could be a nice web extension, @kulmann @JammingBen @AlexAndBear what do you think?

@kobergj FYI

**Comment by dwhite5914** [comment]
@micbar Makes sense. I think it's just how we run things in our lab that would make this useful. For context, it's a small group where everyone knows each other pretty well. Admins generally have ultimate oversight over all of our other services. And we have tight access controls that admins receive special training for and are responsible for enforcing, so we generally need access to enforce these boundaries in the event someone unintentionally gives access they shouldn't.

**Comment by JammingBen** [comment]
It could be solved via a web-extension, but we don't have an extension point for it and I'm struggling to find a generic use case to be honest. It would be something like a "post create space hook" to handle additional stuff after a space has been created.

Also, how does the creator of a space retrieve all the ids of the admin users for the invite requests? Is that information even accessible for a non-admin?

All in all I'm not convinced it should be a web extension. Maybe @dschmidt has some (different) thoughts about it?

**Comment by JammingBen** [comment]
> My idea was an extension which hooks into the „create space“ action button. But not a post-create, I thought of a pre create to present a share dialogue like in the sidebar to the user where they can add members.

But then users would still need to select each member manually when creating a space. I thought the desired solution was something that automatically adds a predefined set of users/groups 🤔 

Either way though, I still think it's not generic enough to justify a web extension (point). This rather is a specific feature request.

**Comment by kobergj** [comment]
@scmanjarrez currently there is no option to add a default user to every space. Managers need to be assigned manually when the space is created. We could explore the option of having a `default space manager` - a userID that automatically gets added to all spaces. But this is currently not on any roadmap.

If you mean by "super administrator" a user that has access to all spaces without being added, I'm afraid this is not possible. Data in spaces is considered private to the members of the space. No user can access the files if they don't have access to the space. This however can bring problems like orphaned spaces - spaces that do not have a manager any more. There is a proposal how to fix this https://github.com/owncloud/ocis/issues/5967 but it was not greenlit yet. The workaround is to download all files from an orphaned space (viewer permission needed at least) and reupload them to a new space.

**Comment by scmanjarrez** [comment]
I didn't mean to access the files themselves, but rather to manage the space's settings, such as increasing or decreasing a member's permissions (or even modifying the membership list). I understand the policy of treating a space as a private container, but if these actions can be performed by directly accessing OCIS files on the server, then providing the same functionality within the Admin dashboard seems reasonable.

Are the files encrypted on the server? If not, there's nothing stopping a malicious "super administrator" from accessing private files within a space anyway.

I was thinking of the administrator role in a GitLab instance (which is usually only 1 person), for example, where you can modify any system parameter. This account is typically used only in very specific situations. The same parameters can also be modified via a CLI tool on the server, so the GUI offers these features primarily as a QoL improvement.

**Comment by kobergj** [comment]
> if these actions can be performed by directly accessing OCIS files on the server, then providing the same functionality within the Admin dashboard seems reasonable.

This is not always true. The sysadmins deploying the software could be different persons than the ocis admins. Only in very small installation this will be the same person. 

> Are the files encrypted on the server? 

That depends on the installation. Ocis supports encrypted file system so this could be possible. If your sysadmins do not have access to the encryption keys they already cannot access any file any more.

> I was thinking of the administrator role in a GitLab instance

I see. This would be a role that can change the space metadata (name, description, quota, members) but not see the files itself. We thought about but this brings new problems. What stops the superadmin to add itself to a space? Even if we would block this the superadmin could still use a second user to hijack spaces. Also the superadmin could remove users from spaces to start a dos attack. 
So as you see while a superadmin does solve some problems, it also brings a lot of new ones. 

> Your linked issue seems as a good solution.

I thought so too but the ticket was not accepted yet. Probably needs more refining as it is too complicated to set up.
